.travis.yml

```diff
@@ -62,15 +59,44 @@ jobs:
           repo: pytest-dev/pytest
     - stage: linting
       python: '3.6'
-      env:
+      env: PYTEST_NO_COVERAGE=1
       install:
       - pip install pre-commit
       - pre-commit install-hooks
       script:
       - pre-commit run --all-files
 
+before_script:
+  - |
+    if [[ "$PYTEST_NO_COVERAGE" != 1 ]]; then
+      set -ex
+      # Massage .coveragerc to use source from tox env (only).
+      sed -i "s~source~source = $(ls $PWD/.tox/$TOXENV/lib/python*/site-packages/_pytest/),$PWD/testing" .coveragerc
```

Comment:
one trick you can use here is `.coveragerc` supports environment variables.

I'm trying to find a not-closed-source example for you but this is the rough sketch:

```ini
setenv =
    TOP={toxinidir}
```

and then in `.coveragerc`

```
source = $TOP
```

(I think `$PWD` probably works fine too, without `setenv` -- though I'm trying to figure out why different 🤔 )

The rough part here is when things `chdir` you want an absolute path...